session/session.go

```diff
@@ -13,11 +13,18 @@ import (
 )
 
 var userIDKey = struct{}{}
+var allowedAnonymousKey = struct{}{}
```

Comment:
nit: can these be structs instead? e.g. `type userIDKey struct {}` and `type allowedAnonymousKey struct {}`, [ex](https://github.com/RMI/opgee-api/blob/d342c0c21c0dbf4c62025a4ee71a74094f4f37f0/opgee/opgee.go#L47-L51)

That's the normal shape I've seen for private context keys, using anonymous structs in this nature seems less solid.

Reply:
SGTM.